WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/management/InsightsManagementViewModel.kt

```diff
@@ -72,9 +72,8 @@ class InsightsManagementViewModel @Inject constructor(
     }
 
     fun onSaveInsights() {
-        // This has to be GlobalScope because otherwise the coroutine gets killed with the ViewModel
         analyticsTrackerWrapper.track(Stat.STATS_INSIGHTS_MANAGEMENT_SAVED)
-        GlobalScope.launch {
+        insightsUseCase.launch(mainDispatcher) {
```

Comment:
Since we want the action to survive the InsightsManagement screen, we can run it in the use case scope which will live long enough to handle it.